File: 0283. Move Zeroes.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        int index = 0;
        for(int i = 0 ; i < nums.Length ; i++){
            int n = nums[i]; 
            // 非0的數先排到前面
            if(n != 0){
                nums[index++] = n;    
            }
        }

        // 排完後把0補上
        for(int i = index ; i < nums.Length ; i++){
            nums[i] = 0;
        }
    }
}

File: 0283. Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        index = 0
        
        for i in range(len(nums)):
            n = nums[i] 
            if n != 0:
                nums[index] = n    
                index = index + 1
        

        
        for j in range(index, len(nums)):
            nums[j] = 0;
        